daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/parsers/ElementCombinator1.scala

```diff
@@ -198,8 +198,11 @@ abstract class ElementParserBase(
 
       if (pstate.processorStatus ne Success) return
 
+      // if we should do limited validation via CheckConstraints
       val shouldValidate = erd.isSimpleType &&
-        pstate.dataProc.isDefined && pstate.dataProc.value.validationMode != ValidationMode.Off
+        pstate.dataProc.isDefined &&
+        pstate.dataProc.value.validationMode != ValidationMode.Off &&
+        pstate.dataProc.value.validationMode != ValidationMode.Full
```

Comment:
Suggest we just do

```scala
pstate.dataProc.isDefined && pstate.dataProc.value.validationMode == ValidationMode.Limited
```
So a single positive test instead of a bunch of negative tests. If we use negative checks and ever add more validation modes the list of != could get kindof big. And this way if we ever add more validation modes, this won't have to change since this should only ever be enabled with limited validation.